SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggRoutines.h

```diff
@@ -45,7 +45,7 @@ void picnic_at_zero_gate(const ProgramInfo& info, ConsoleHandle& console, BotBas
 // - Two-sweet-herbs, bitter-sweet-herbs or salty-sweet-herbs custom sandwich to gain egg power Lv 3, must have enough ingredinets and
 //   provide game language for OCR ingredient lists.
 // Return false if no needed sandwich ingredients or recipe.
-bool eat_egg_sandwich_at_picnic(const ProgramInfo& info, AsyncDispatcher& dispatcher, ConsoleHandle& console,
+bool eat_egg_sandwich_at_picnic(SingleSwitchProgramEnvironment& env, AsyncDispatcher& dispatcher, ConsoleHandle& console,

```

Comment:
I'm a bit iffy about this change since it means we can't call this function in a multi-Switch program. But at the same time, we currently don't have any multi-Switch programs that make sandwiches.

Looking further down, it looks like the original sandwich maker already took `SingleSwitchProgramEnvironment` as a parameter. Tracing it further down, the only part that it uses is log and the overlay - neither of which are special to `SingleSwitchProgramEnvironment`.

The "proper" thing to do here (which should've been done in the beginning a long time ago), is to replace `SingleSwitchProgramEnvironment& env` with `const ProgramInfo& info` and `ConsoleHandle& console`. Since `console` is already passed, it looks like the best thing here is to leave it as `const ProgramInfo& info`. Then all logging calls `env.log(...)` become `console.log(...)` and all the overlay calls `env.console.overlay()` become just `console.overlay()`.

Reply:
So same as above:

`SingleSwitchProgramEnvironment -> ProgramEnvironment`

and remove `AsyncDispatcher& dispatcher` from parameter since it's inside `env`.